### PR TITLE
Application Gateway for Containers - PowerShell

### PR DESCRIPTION
**Pull request by JackStromberg**
## Description

PowerShell for Application Gateway for Containers.  Using ALB name in consistency with CLI, Terraform, and k8s implementations.

## Checklist

- [ x] Check this box to confirm: **I have read the [_Submitting Changes_](../blob/main/CONTRIBUTING.md#submitting-changes) section of [`CONTRIBUTING.md`](../blob/main/CONTRIBUTING.md) and reviewed the following information:**

* **SHOULD** select appropriate branch. Cmdlets from Autorest.PowerShell should go to [`generation`](https://github.com/Azure/azure-powershell/tree/generation) branch. 
* **SHOULD** make the title of PR clear and informative, and in the present imperative tense. 
* **SHOULD** update `ChangeLog.md` file(s) appropriately
    * For any service, the `ChangeLog.md` file can be found at `src/{{SERVICE}}/{{SERVICE}}/ChangeLog.md`
    * A snippet outlining the change(s) made in the PR should be written under the `## Upcoming Release` header in the past tense. Add changelog in description section if PR goes into [`generation`](https://github.com/Azure/azure-powershell/tree/generation) branch.
    * Should **not** change `ChangeLog.md` if no new release is required, such as fixing test case only.
* **SHOULD** have approved design review for the changes in [this repository](https://github.com/Azure/azure-powershell-cmdlet-review-pr) ([_Microsoft internal only_](../blob/main/CONTRIBUTING.md#onboarding)) with following situations
    * Create new module from scratch
    * Create new resource types which are not easy to conform to [Azure PowerShell Design Guidelines](../blob/main/documentation/development-docs/design-guidelines)
    * Create new resource type which name doesn't use module name as prefix
    * Have design question before implementation
* **SHOULD** regenerate markdown help files if there is cmdlet API change. [Instruction](../blob/main/documentation/development-docs/help-generation.md#updating-all-markdown-files-in-a-module)
* **SHOULD** have proper test coverage for changes in pull request.
* **SHOULD NOT** introduce [breaking changes](../blob/main/documentation/breaking-changes/breaking-changes-definition.md) in Az minor release except preview version.
* **SHOULD NOT** adjust version of module manually in pull request

### CODE REVIEW

**Comment by Nickcandy** [comment]
@JackStromberg Hi Jack, please fix the [example issue](https://dev.azure.com/azclitools/public/_build/results?buildId=73038&view=logs&j=b9afb22f-0837-5d02-bc24-6c30aee08bcf&t=990f6f37-4a21-5927-092a-220f38a460e9) in the ci pipeline:

File: src/ALB/README.md [hunk 1]
```diff
@@ -0,0 +1,167 @@
...
+module-version: 0.2.0
+title: ALB
+subject-prefix: $(service-name)
+inlining-threshold: 100
+resourcegroup-append: true
+nested-object-to-string: true
+
+# pin the swagger version by using the commit id instead of branch name
+#commit: 4ff04ab2f7c19a83ca1c83d7189238930ca0d1ae # testing
+#commit: e7fa731fdc72d810a74bf4c83184e3d64661067e # testing
```
Comment (wyunchi-ms) [inline src/ALB/README.md:43]: Please remove the commentted commits

File: src/ALB/README.md [hunk 2]
```diff
@@ -0,0 +1,167 @@
+<!-- region Generated -->
+# Az.Alb
+This directory contains the PowerShell module for the Alb service.
+
+---
+## Status
+[![Az.Alb](https://img.shields.io/powershellgallery/v/Az.Alb.svg?style=flat-square&label=Az.Alb "Az.Alb")](https://www.powershellgallery.com/packages/Az.Alb/)
```
Comment (wyunchi-ms) [inline src/ALB/README.md:7]: Please all use `ALB` instead of `Alb`

File: src/ALB/README.md [hunk 3]
```diff
@@ -0,0 +1,164 @@
...
```
Comment (Nickcandy) [inline src/ALB/README.md:7]: Please change all your ALB to Alb including the module folder. Follow this branch: https://github.com/Azure/azure-powershell/pull/22359/files

File: src/ALB/README.md [hunk 4]
```diff
@@ -0,0 +1,164 @@
...
+## Development
+For information on how to develop for `Az.Alb`, see [how-to.md](how-to.md).
+<!-- endregion -->
+
+### AutoRest Configuration
+> see https://aka.ms/autorest
+
+```yaml
+
+module-version: 0.2.0
```
Comment (Nickcandy) [inline src/ALB/README.md:34]: If this a new module, use 0.1.0

File: src/ALB/README.md [hunk 5]
```diff
@@ -0,0 +1,164 @@
...
+      subject: Associations
+    set:
+      subject: Association
+  - where: 
+      subject: Frontends
+    set: 
+      subject: Frontend
+# Renames for parameters continued
+  - where:
+      subject: Frontend
```
Comment (Nickcandy) [inline src/ALB/README.md:104]: Please make sure you've changed the parameter name after building.

File: src/ALB/changelog.md [hunk 6]
```diff
@@ -0,0 +1,15 @@
+## Upcoming Release
```
Comment (Nickcandy) [inline src/ALB/changelog.md:1]: You should remove the changelog here and just leave the comments. We would add changelog for you in main branch.

File: src/ALB/test/Get-AzAlb.Tests.ps1 [hunk 7]
```diff
@@ -0,0 +1,33 @@
...
+  $TestRecordingFile = Join-Path $PSScriptRoot 'Get-AzAlb.Recording.json'
+  $currentPath = $PSScriptRoot
+  while(-not $mockingPath) {
+      $mockingPath = Get-ChildItem -Path $currentPath -Recurse -Include 'HttpPipelineMocking.ps1' -File
+      $currentPath = Split-Path -Path $currentPath -Parent
+  }
+  . ($mockingPath | Select-Object -First 1).FullName
+}
+
+Describe 'Get-AzAlb' {
```
Comment (Nickcandy) [inline src/ALB/test/Get-AzAlb.Tests.ps1:17]: Please fill up your tests. Maybe not all the parametersets but most of them.

**Comment by Nickcandy** [comment]
@JackStromberg Since the tests have not ready, we recommend that we could postpone the ALB module creation to the next release if there is no requirement.

File: src/Alb/ChangeLog.md [hunk 8]
```diff
@@ -0,0 +1,14 @@
+* Initial release - public preview
```
Comment (Nickcandy) [inline src/Alb/ChangeLog.md:1]: Please delete changelog here

**Review by Nickcandy — APPROVED** [review]

**Pull request merged**